### DISCUSSION TITLE
Moving sprites on the screen are blurry.

### DISCUSSION THREAD

**Discussion by Darenn** [nesbox/TIC-80#1833] [Q&A] [answered]
Hi everyone,

As you can see in the video, non moving sprites like the player and the HUD are displayed correctly.
But the sprites that are moving on the screen every frame are blurry. It's really obvious if you focus your attention on a planet while moving.

I'm sure I might be doing something wrong, but can't figure out what.

Here's the code drawing the planets, it's called every tick.
`spr(planet.spr, planet.x-cam.x, planet.y-cam.y, 0, 1, 0, 0, planet.spr_size, planet.spr_size)`

At first I thought it was maybe a bad idea to draw every frame, but as expected it changed nothing.
Then I made sure the camera coordinates were all integer, it fixed other position issues but not the blur.

Thanks a lot for you help, don't hesitate to ask if you need more info.

https://user-images.githubusercontent.com/11272715/152814159-c2df88e0-4dd0-466b-86b7-fc0d31454307.mp4

Edit : here's a link to the game, it's actually hard to see it on video : https://darenn.itch.io/last-in-space

**Comment by paul59** [comment]
How are you handling movement/speed/input, using timers or just adding some value to the camera position every tick/when input received? TBH it doesn't look too bad to me!

**Reply by Darenn** [reply]
Thanks for your help Paul.
Here's the code of my camera movement :
`	cam.x=p.x+8-SCREEN_WIDTH/2
	cam.y=p.y+8-SCREEN_HEIGHT/2`
I call this every frames. The player is moved every two frames (to allow me to move at slow speed like 30 pixel per second).

Just to be sure It's not because I give floating values I tried moving my player by one pixel every frame in the TIC.
`p.x = p.x + 1
p.y = p.y + 1`
And it gives the exact same effect.

**Comment by joshgoebel** [comment]
Yes, it looks fine to me also.  I don't see any blurring.

**Reply by Darenn** [reply]
Thanks for looking into it Josh. If you keep looking at a planet during a movement, you don't see some blur ? While if you look at the player there's no problem at all.

**Comment by joshgoebel** [comment]
You can literally step thru the video frame by frame, there is no blur.

**Reply by Darenn** [reply]
You're right, I see the blur only while it's moving fast.
Maybe something with my screen ? The strange thing is that I see it on my second monitor and on another computer. Maybe I need to see a doctor. :D

Do you see it when playing ? : https://darenn.itch.io/last-in-space

**Reply by joshgoebel** [reply]
I think it's just par for the course, fast moving objects can sometimes look blurry... if the video capture is clean then the image being sent to the screen is clean... so it's just your eyes (and potentially your display hardware) that's left to blame.

It's important to note **your ship isn't moving** - it's largely stationary (right smack in the middle)... so it's not a fair comparison.

**Reply by Darenn** [reply]
Yes, I've compared with the ship because its not moving and looks ok. Would like every sprites to look like that. You're right it's definitely a problem with the movement of the images and not each frame itself.

I've looked on another computer, and maybe I'm using the wrong terms. It's not blurry, but it look like the planet is drawn several times, I see kind of a shadow. My friend said it's like it's quickly blinking.

**Reply by mooncalf** [reply]
looks great. good start on the game too, looks cool. i'm pretty sure all you're seeing is the normal jitter from pixels and eyeballs, man! looks fine. the ship is sitting still so it's not tripping your eyes out, but those planets are moving tiny little jumps... you're going to see that on any pixelated game.

**Reply by paul59** [reply]
@Darenn I understand what you mean but it's a facet of pixel based games on large screens. A TIC pixel movement of +1 will equate to many pixels on a typical monitor which makes it look jumpy compared to when you drag a window around  for example.

You could try reducing the speed (but keeping the ratio between high and low speeds) so planets move less often or try easing the camera to the player position rather than jumping to it.

**Reply by Darenn** [reply]
@mooncalf  thanks for your kind words.
Thanks for you help everyone!

I noticed something. When I don't move, but sprites are moving on the screen, I don't see the effect at all!
So yeah it's definitely a problem with how I handle the camera

**Comment by Darenn** [comment, marked as answer]
Hey everyone,
Thanks again for your help.
I changed my camera to a smooth camera, and added to my player a "logic position" and "draw position".
The logic one is decimal, and used of speed and things. The draw one is integer and used to draw at the right position (rounding the logic position).
Playing around with the speed of the camera and player, I managed to get something where I don't get the visual effect anymore.
Seems like I was just big pixels too fast for our eyes.

Closing!
